conn.go

```diff
@@ -69,9 +69,11 @@ type Conn struct {
 	rootCAs               *x509.CertPool
 	serverName            string
 
-	handshakeMessageHandler handshakeMessageHandler
-	flightHandler           flightHandler
-	handshakeCompleted      chan bool
+	handshakeMessageHandler             handshakeMessageHandler
+	flightHandler                       flightHandler
+	handshakeCompletedSignal            chan bool
+	handshakeCompletedSuccessfully      bool
```

Comment:
mind using an atomic instead, just one less member and we don't need to worry about invalid access!

Reply:
Yes, good idea :+1: